disnake/flags.py

```diff
@@ -59,15 +62,24 @@
 )
 
 BF = TypeVar("BF", bound="BaseFlags")
+T = TypeVar("T", bound="BaseFlags")
 
 
-class flag_value:
+class flag_value(Generic[T]):
     def __init__(self, func: Callable[[Any], int]):
         self.flag = func(None)
         self.__doc__ = func.__doc__
+        self._parent: Type[T] = MISSING
+
+    def __or__(self, other: flag_value[T]) -> T:
+        assert self._parent is other._parent  # noqa: S101
```

Comment:
this should raise a proper exception instead of asserting
same thing with the `BaseFlags` changes further down

Reply:
https://github.com/DisnakeDev/disnake/pull/615/commits/9f2838f94bfde5f92d6b247e36b2033cddfa8f34